amx/server/syscalls.lua

```diff
@@ -2765,53 +2765,59 @@ end
 
 -- Now we have all of RESTful types of requests. Our function is better!
 -- The SAMP documentation said about 'url' - "The URL you want to request. (Without 'http://')"
--- But I didn't set it up. Because WE SUPPORT HTTPS!
+-- I made a check. The state without a protocol is called as 'default'.
+-- HTTP and HTTPS you can put into URL if you want. It works fine.
 -- TODO: An "index" argument only for compatibility.
 function HTTP(amx, index, type, url, data, callback)
 
+	local protomatch = pregMatch(url,'^(\\w+):\\/\\/')
+	local proto = protomatch[1] or 'default'
+	-- if somebody will try to put here ftp:// ssh:// etc...
+	if proto ~= 'http' and proto ~= 'https' and proto ~= 'default' then
+		print('Current protocol is not supporting')
+		return 0
+	end
 	local typesToText = {
 		'GET',
 		'POST',
 		'HEAD',
-		'PUT',
-		'PATCH',
-		'DELETE',
-		'COPY',
-		'OPTIONS',
-		'LINK',
-		'UNLINK',
-		'PURGE',
-		'LOCK',
-		'UNLOCK',
-		'PROPFIND',
-		'VIEW'
+		[-4] = 'PUT',
+		[-5] = 'PATCH',
+		[-6] = 'DELETE',
+		[-7] = 'COPY',
+		[-8] = 'OPTIONS',
+		[-9] = 'LINK',
+		[-10] = 'UNLINK',
+		[-11] = 'PURGE',
+		[-12] = 'LOCK',
+		[-13] = 'UNLOCK',
+		[-14] = 'PROPFIND',
+		[-15] = 'VIEW'
 	}
 	local sendOptions = {
-		queueName = "A SAMP request queue",
+		queueName = "amx." .. getResourceName(amx.res) .. "." .. amx.name,
 		postData = data,
 		method = typesToText[tonumber(type)],
 	}
-	fetchRemote(url, sendOptions, 
+	local successRemote = fetchRemote(url, sendOptions, 

```

Comment:
This line uses `successRemote` but the return at the end checks for `success`

Reply:
Yes, I fixed it after commit :( 6e1259ac8a5c3f819205bb273ae3723b66c147a6

Reply:
oh oops i even saw that commit and didn't realise it fixes it lol